api/pkg/runner/vllm_runtime.go

```diff
@@ -0,0 +1,410 @@
+
+	// Create a simple request body
+	requestBody := `{
+		"model": "` + model + `",
+		"messages": [
+			{"role": "user", "content": "Say the word 'warm'."}
+		],
+		"max_tokens": 5
+	}`
+
+	// Create the HTTP request
+	req, err := http.NewRequestWithContext(ctx, "POST", url, strings.NewReader(requestBody))
+	if err != nil {
+		return fmt.Errorf("error creating warm-up request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	// Send the request
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("error sending warm-up request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	// Check if the request was successful
+	if resp.StatusCode >= 400 {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("error warming up model, status: %d, response: %s", resp.StatusCode, string(bodyBytes))
+	}
+
+	return nil
+}
+
+func (v *VLLMRuntime) Runtime() types.Runtime {
+	return types.RuntimeVLLM
+}
+
+func (v *VLLMRuntime) Version() string {
+	return v.version
+}
+
+func (v *VLLMRuntime) Status(_ context.Context) string {
+	// vLLM doesn't have a built-in status endpoint like Ollama
+	// For now, just return a simple status
+	return "running"
+}
+
+func (v *VLLMRuntime) waitUntilVLLMIsReady(ctx context.Context, startTimeout time.Duration) error {
+	startCtx, cancel := context.WithTimeout(ctx, startTimeout)
+	defer cancel()
+
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-startCtx.Done():
+			return startCtx.Err()
+		case <-ticker.C:
+			// Try to connect to the vLLM server's health endpoint
+			url := fmt.Sprintf("%s/v1/models", v.URL())
+			req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+			if err != nil {
+				continue
+			}
+
+			client := &http.Client{Timeout: 1 * time.Second}
+			resp, err := client.Do(req)
+			if err != nil {
+				continue
+			}
+			resp.Body.Close()
+
+			if resp.StatusCode < 400 {
+				return nil
+			}
+		}
+	}
+}
+
+func startVLLMCmd(ctx context.Context, commander Commander, port int, cacheDir string, contextLength int64, model string, customArgs []string) (*exec.Cmd, error) {
+	// Find vLLM on the path
+	vllmPath, err := commander.LookPath("python")
+	if err != nil {
+		return nil, fmt.Errorf("python not found in PATH")
+	}
+	log.Debug().Str("python_path", vllmPath).Msg("Found python")
+
+	// Prepare vLLM serve command
+	log.Debug().Msg("Preparing vLLM serve command")
+
+	// First prepare a map of custom arguments for quick checking
+	customArgsMap := make(map[string]bool)
+	for i, arg := range customArgs {
+		if i < len(customArgs)-1 && strings.HasPrefix(arg, "--") {
+			customArgsMap[arg] = true
+		}
+	}
+
+	// Build base arguments
+	args := []string{
+		"-m", "vllm.entrypoints.openai.api_server",
+	}
+
+	// Only add default arguments if they're not overridden in customArgs
+	if !customArgsMap["--host"] {
+		args = append(args, "--host", "127.0.0.1")
```

Comment:
This probably wants to be 0.0.0.0

Reply:
0.0.0.0 makes more sense as a bind address, I think we can safely assume we have IPv4 loopback!